.github/workflows/docs-and-linting.yml

```diff
@@ -9,7 +9,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        go: ['1.19', '1.20', '1.21']
+        go: ['1.21', '1.22']
```

Comment:
IMO we should add back 1.18 (matching go.mod), but I don't feel strongly enough to block if other maintainers disagree

I recall there's a way we could have setup-go read go.mod directly too, if we wanted to get fancy, but I don't think that's strictly necessary either